assignment_1/src/State.java

```diff
@@ -5,75 +5,96 @@
-		this.agentX = agentX;
-		this.agentY = agentY;
-		this.score = score;
-		this.turn = turn;
-		this.food = food;
-	}
+    public State(char[][] board, int[] agentX, int[] agentY, int[] score, int turn, int food) {
+        this.board = board;
+        this.agentX = agentX;
+        this.agentY = agentY;
+        this.score = score;
+        this.turn = turn;
+        this.food = food;
+    }
 
-	public void read(String file) throws IOException{
-		File f = new File(file);
-		Scanner reader = new Scanner(f);
-		String[] hw = reader.nextLine().split(" ");
-		int h = Integer.parseInt(hw[0]);
-		int w = Integer.parseInt(hw[1]);
-		agentX = new int[2];
-		agentY = new int[2];
-		board = new char[w][h];
-		food = 0;
-		turn = 0;
-		score = new int[0];
+    // A new constructor that will take in a State object that you want to copy and create a brand new object.
+    public State(State state) {
+        this.copy(state);
+    }
 
-		try {
-			while(reader.hasNextLine()){
-				for (int i = 0; i < w; i++) {
-					char[] s = reader.nextLine().toCharArray();
-					for (int j = 0; j < s.length; j++) {
-						board[i][j] = s[j];
-					}
-				}
-			}
-		} catch (NoSuchElementException e) {
-			e.printStackTrace();
-		}
+    public void read(String file) throws IOException {
+        File f = new File(file);
+        Scanner reader = new Scanner(f);
+        String[] hw = reader.nextLine().split(" ");
+        int h = Integer.parseInt(hw[0]);
+        int w = Integer.parseInt(hw[1]);
+        agentX = new int[2];
+        agentY = new int[2];
+        board = new char[w][h];
+        food = 0;
+        turn = 0;
+        score = new int[0];
 
-		reader.close();
+        try {
+            while (reader.hasNextLine()) {
+                for (int i = 0; i < w; i++) {
+                    char[] s = reader.nextLine().toCharArray();
+                    for (int j = 0; j < s.length; j++) {
+                        board[i][j] = s[j];
+                    }
+                }
+            }
+        } catch (NoSuchElementException e) {
+            e.printStackTrace();
+        }
 
-		for(char[] r: board) {
-			for(char c: r){
-				if(c == '*') { food++; }
-			}
-		}
-	}
+        reader.close();
 
-	public String toString(){
-		StringBuilder s = new StringBuilder();
-		// Loop through all rows
-		for (int i = 0; i < board.length; i++)
-			for (int j = 0; j < board[i].length; j++)
-				s.append(board[i][j]).append(" ");
-		return s.toString();
-	}
+        for (char[] r : board) {
+            for (char c : r) {
+                if (c == '*') {
+                    food++;
+                }
+            }
+        }
+    }
 
-	public boolean isLeaf() {
-		return true;
-	}
+    /*
+        Create a copy method that will be used for a new copy constructor that will use this method to assign the fields
+        from the object you want to copy (state.x) to your new object (this.x), fairly Straight forward.
+    */
+    public void copy(State state) {
```

Comment:
It is specified in the assignment that the method should return a new State object so the method should look like this:

```suggestion
    public State copy(State state) {
```